spec/formatting.md

```diff
@@ -18,3 +18,71 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Error Handling
+
+During the formatting of a message,
+various errors may be encountered.
+These are divided into the following categories:
+
+- **Syntax errors** occur when the syntax representation of a message is invalid.
+- **Resolution errors** occur when the runtime value of a part of a message
+  cannot be determined.
+
+  - **Unresolved Variable errors** occur when a variable reference cannot be resolved.
+
+- **Selection errors** occur when message selection fails.
+
+  - **Selector errors** are failures in the matching of a key to a specific selector.
+  - **Missing Fallback errors** occur when no Variant is selected
+    due to the message not including a Variant with only catch-all keys.
+
+- **Formatting errors** occur during the formatting of a resolved value,
+  for example when encountering a value with an unsupported type
+  or an internally inconsistent set of options.
+
+During selection, an expression handler must only emit Resolution and Selection errors.
+During formatting, an expression handler must only emit Resolution and Formatting errors.
+
+In all cases, when encountering an error,
+a message formatter must provide some representation of the message.
+An informative error or errors must also be separately provided.
+
+When an error occurs in the syntax or resolution of an Expression or MarkupStart Option,
+the Expression or MarkupStart in question is processed as if the option were not defined.
```

Comment:
Should not mention MarkupStart, as there is no agreement on markup.

Reply:
Are objecting here out of principle, or do you specifically think that failures in Expression and Markup options should be handled differently?